test/sagemaker_tests/pytorch/training/resources/mnist/smdataparallel_mnist.py

```diff
@@ -23,15 +24,23 @@
 from packaging.version import Version
 from torchvision import datasets, transforms
 from torch.optim.lr_scheduler import StepLR
-from torch.nn.parallel import DistributedDataParallel as DDP
-import torch.distributed as dist
-import smdistributed.dataparallel.torch.torch_smddp
 
-# set default instance type to p3.16
-if "SAGEMAKER_INSTANCE_TYPE" not in os.environ:
-    os.environ['SAGEMAKER_INSTANCE_TYPE'] = 'ml.p3.16xlarge'
+TORCH_VERSION = torch.__version__
 
-dist.init_process_group(backend='smddp')
+if Version(TORCH_VERSION) < Version("1.10"):
+    from smdistributed.dataparallel.torch.parallel.distributed import DistributedDataParallel as DDP
+    import smdistributed.dataparallel.torch.distributed as dist
```

Comment:
do we need a corresponding import for `import smdistributed.dataparallel.torch.torch_smddp`

Reply:
+1

Reply:
Changes made during PT1.10 release turned the following script incompatible with PT1.9 version. So, the changes that I have made in this file are to make the script backward compatible. To do that, based on the DataParallel team's suggestion, I looked into the script that existed when PT1.9 was released (https://github.com/shantanutrip/deep-learning-containers/blob/b87590af021a4c3913b3669fa5f9fc10fbebe4e8/test/sagemaker_tests/pytorch/training/resources/mnist/smdataparallel_mnist.py) and found its diff with the current file  on the master branch (https://github.com/aws/deep-learning-containers/blob/master/test/sagemaker_tests/pytorch/training/resources/mnist/smdataparallel_mnist.py) and made the changes accordingly. 

For the old file, I do not see any `import smdistributed.dataparallel.torch.torch_smddp` line or any other import corresponding to it, so I did not add it.

Reply:
Older SMDDP versions don't need an equivalent import